homeassistant/components/mpd/media_player.py

```diff
@@ -73,6 +74,7 @@
         vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
         vol.Optional(CONF_PASSWORD): cv.string,
         vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+        vol.Optional(CONF_UNIQUE_ID): cv.string,
```

Comment:
We no longer accept additions or changes to the yaml config of integrations. See: https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md

This integration needs to be refactored to use a config flow and config entries.

https://developers.home-assistant.io/docs/config_entries_config_flow_handler

Reply:
We also don't generally allow integrations to configure unique id. That should be set by device data like serial number or Mac address.

Reply:
Hey Martin, I understand about modifying the YAML and config flows. I don't have the time to add a config flow for this integration. 

Would a PR for automatically setting the unique ID to `mpd-{name}-{hostname}:{port}` be accepted?

Reply:
The unique_id must both be unique and stable. We don't allow it to be based on user configuration. Ip address is also not allowed as that can change.

Reply:
https://developers.home-assistant.io/docs/entity_registry_index/#unique-id-requirements

Reply:
I understand. I'll close the PR.